webapi/src/test/scala/org/knora/webapi/util/AkkaHttpUtils.scala

```diff
@@ -0,0 +1,41 @@
+package org.knora.webapi.util
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.{HttpResponse, StatusCodes}
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.ActorMaterializer
+import akka.util.Timeout
+import spray.json._
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, Future}
+
+/**
+  * Object containing methods for dealing with [[HttpResponse]]
+  */
+object AkkaHttpUtils {
+
+    /**
+      * Given an [[HttpResponse]] containing json, return the said json.
+      * @param response the [[HttpResponse]] containing json
+      * @return an [[JsObject]]
+      */
+    def httpResponseToJson(response: HttpResponse)(implicit ec: ExecutionContext, system: ActorSystem): JsObject = {
+
+        import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+        import DefaultJsonProtocol._
+
+        implicit val materializer = ActorMaterializer()
+
+        val jsonFuture: Future[JsObject] = response match {
+            case HttpResponse(StatusCodes.OK, _, entity, _) =>
+                Unmarshal(entity).to[JsObject]
+            case other =>
+                throw new Exception(other.toString())
+        }
+
+        //FIXME: There is probably a better non blocking way of doing it.
```

Comment:
I'm not sure. Some tests need the json object, to check the response. I will need to read up on this one.